decred/tests/unit/dcr/test_vsp_unit.py

```diff
@@ -0,0 +1,447 @@
+"""
+Copyright (c) 2020, the Decred developers
+See LICENSE for details
+"""
+
+import time
+
+import pytest
+
+from decred import DecredError
+from decred.dcr import vsp
+from decred.dcr.nets import mainnet
+from decred.util import encode
+
+
+def test_result_is_success():
+    class test:
+        def __init__(
+            self, res, isSuccess,
+        ):
+            self.res = res
+            self.isSuccess = isSuccess
+
+    tests = [
+        test({"status": "success"}, True,),
+        test({"status": "fail"}, False,),
+        test({}, False,),
+        test("abcd", False,),
+    ]
+    for test in tests:
+        assert vsp.resultIsSuccess(test.res) is test.isSuccess
+
+
+purchaseInfo = {
+    "PoolAddress": "TsbyH2p611jSWnvUAq3erSsRYnCxBg3nT2S",
+    "PoolFees": 0.5,
+    "Script": "512103af3c24d005ca8b755e7167617f3a5b4c60a65f8318a7fcd1b0cacb1ab"
+    "d2a97fc21027b81bc16954e28adb832248140eb58bedb6078ae5f4dabf21fde5a8ab7135c"
+    "b652ae",
+    "TicketAddress": "Tcbvn2hiEAXBDwUPDLDG2SxF9iANMKhdVev",
+    "VoteBits": 5,
+    "VoteBitsVersion": 0,
+}
+
+
+def test_purchase_info_parse():
+
+    now = int(time.time())
+    pi = vsp.PurchaseInfo.parse(purchaseInfo)
+
+    assert pi.poolAddress == "TsbyH2p611jSWnvUAq3erSsRYnCxBg3nT2S"
+    assert pi.poolFees == 0.5
+    assert (
+        pi.script
+        == "512103af3c24d005ca8b755e7167617f3a5b4c60a65f8318a7fcd1b0cacb1abd2a"
+        "97fc21027b81bc16954e28adb832248140eb58bedb6078ae5f4dabf21fde5a8ab7135"
+        "cb652ae"
+    )
+    assert pi.ticketAddress == "Tcbvn2hiEAXBDwUPDLDG2SxF9iANMKhdVev"
+    assert pi.voteBits == 5
+    assert pi.voteBitsVersion == 0
+    assert isinstance(pi.unixTimestamp, int) and pi.unixTimestamp >= now
+
+
+def test_purchase_info_blobbing():
+
+    pi = vsp.PurchaseInfo.parse(purchaseInfo)
+    stamp = pi.unixTimestamp
+    b = vsp.PurchaseInfo.blob(pi)
+    assert isinstance(b, bytearray)
+
+    rePi = vsp.PurchaseInfo.unblob(b)
+    assert rePi.poolAddress == "TsbyH2p611jSWnvUAq3erSsRYnCxBg3nT2S"
+    assert rePi.poolFees == 0.5
+    assert (
+        rePi.script
+        == "512103af3c24d005ca8b755e7167617f3a5b4c60a65f8318a7fcd1b0cacb1abd2a"
+        "97fc21027b81bc16954e28adb832248140eb58bedb6078ae5f4dabf21fde5a8ab7135"
+        "cb652ae"
+    )
+    assert rePi.ticketAddress == "Tcbvn2hiEAXBDwUPDLDG2SxF9iANMKhdVev"
+    assert rePi.voteBits == 5
+    assert rePi.voteBitsVersion == 0
+    assert rePi.unixTimestamp == stamp
```

Comment:
This object comparison is also done in `test_purchase_info_parse`. A function could be used for this part. You should also compare `rePi` directly to `pi` instead of copying the values around.